Sources/DebugExtensions/DebugProxy+Export.swift

```diff
@@ -0,0 +1,233 @@
+        let fileURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(filePrefix)_image").appendingPathExtension(codec.fileExtension)
+        self.write(to: fileURL, codec: codec, format: format, quality: quality, colorSpace: colorSpace)
+        Task {
+            await openShareSheet(for: [fileURL])
+        }
+    }
+
+}
+
+public extension CIImage.DebugProxy.RenderResult {
+
+    /// Opens a share sheet for exporting the rendering artifacts (the image as TIFF and various rendering graphs as PDFs).
+    /// On macOS, the system dialog for picking an export folder will be shown instead.
+    /// - Parameter filePrefix: A prefix that is added to the files. By default, the name of the app is used.
+    func export(filePrefix: String? = nil) {
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        let shareItems = self.writeResults(to: FileManager.default.temporaryDirectory, with: filePrefix)
+        Task {
+            await openShareSheet(for: shareItems)
+        }
+    }
+
+}
+
+/// Opens a share sheet for exporting the given files from the application's main window.
+/// - Parameter items: A list of files to export. The files will be either moved or deleted after successful export.
+@MainActor private func openShareSheet(for items: [URL]) {
+    let window = UIApplication.shared.windows.first
+
+    if #available(iOS 14, *), ProcessInfo().isMacCatalystApp || ProcessInfo().isiOSAppOnMac {
+        let documentPicker = UIDocumentPickerViewController(forExporting: items)
+        window?.rootViewController?.present(documentPicker, animated: true)
+    } else {
+        let activityViewController = UIActivityViewController(activityItems: items, applicationActivities: nil)
+        activityViewController.completionWithItemsHandler = { _, _, _, _ in
+            items.forEach { try? FileManager.default.removeItem(at: $0) }
+        }
+        activityViewController.popoverPresentationController?.sourceView = window
+        activityViewController.popoverPresentationController?.sourceRect = CGRect(x: window?.bounds.midX ?? 0, y: 0, width: 1, height: 20)
+        window?.rootViewController?.present(activityViewController, animated: true)
+    }
+}
+
+#endif // UIKit
+
+
+// MARK: - Common
+
+extension CIImage.DebugProxy {
+
+    public enum ImageCodec {
+        case exr
+        case jpeg
+        case heif
+        @available(iOS 15, macOS 12, macCatalyst 15, *)
+        case heif10
+        case png
+        case tiff
+
+        fileprivate var fileExtension: String {
+            switch self {
+                case .exr: return "exr"
+                case .jpeg: return "jpeg"
+                case .heif, .heif10: return "heic"
+                case .png: return "png"
+                case .tiff: return "tiff"
+            }
+        }
+    }
+
+    private func write(to fileURL: URL, codec: ImageCodec, format: CIFormat = .RGBA8, quality: Float = 1.0, colorSpace: CGColorSpace?) {
```

Comment:
Is 1.0 quality a good default or could it be like 0.8? Does it even matter for debug though...